pkg/storage/chunk/client/baidubce/bos_storage_client.go

```diff
@@ -0,0 +1,173 @@
+}
+
+// RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
+func (cfg *BosStorageConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&cfg.BucketName, prefix+"baidubce.bucket-name", "", "Name of Bos bucket.")
+	f.StringVar(&cfg.Endpoint, prefix+"baidubce.endpoint", DefaultEndpoint, "Bos endpoint to connect to.")
+	f.StringVar(&cfg.AccessKeyID, prefix+"baidubce.access-key-id", "", "Baidu BCE Access Key ID")
+	f.StringVar(&cfg.SecretAccessKey, prefix+"baidubce.secret-access-key", "", "Baidu BCE Secret Access Key")
+}
+
+type BosObjectStorage struct {
+	cfg    *BosStorageConfig
+	client *bos.Client
+}
+
+func (cfg *BosStorageConfig) Validate() error {
+	return nil
+}
+
+func NewBosObjectStorage(cfg *BosStorageConfig) (*BosObjectStorage, error) {
+	clientConfig := bos.BosClientConfiguration{
+		Ak:               cfg.AccessKeyID,
+		Sk:               cfg.SecretAccessKey,
+		Endpoint:         cfg.Endpoint,
+		RedirectDisabled: false,
+	}
+	bosClient, err := bos.NewClientWithConfig(&clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &BosObjectStorage{
+		cfg:    cfg,
+		client: bosClient,
+	}, nil
+}
+
+func (b *BosObjectStorage) PutObject(ctx context.Context, objectKey string, object io.ReadSeeker) error {
+	return instrument.CollectedRequest(ctx, "Bos.PutObject", bosRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		body, err := bce.NewBodyFromSizedReader(object, -1)
+		if err != nil {
+			return err
+		}
+		_, err = b.client.BasicPutObject(b.cfg.BucketName, objectKey, body)
+		return err
+	})
+}
+
+func (b *BosObjectStorage) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, int64, error) {
+	var res *api.GetObjectResult
+	err := instrument.CollectedRequest(ctx, "Bos.GetObject", bosRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		var requestErr error
+		res, requestErr = b.client.BasicGetObject(b.cfg.BucketName, objectKey)
+		return requestErr
+	})
+	if err != nil {
+		return nil, 0, errors.Wrap(err, "failed to get bos object")
+	}
+	var size int64
+	if res.ContentLength != 0 {
+		size = res.ContentLength
+	}
+	return res.Body, size, nil
+}
+
+func (b *BosObjectStorage) List(ctx context.Context, prefix string, delimiter string) ([]client.StorageObject, []client.StorageCommonPrefix, error) {
+	var storageObjects []client.StorageObject
+	var commonPrefixes []client.StorageCommonPrefix
+
+	err := instrument.CollectedRequest(ctx, "Bos.List", bosRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
+		args := new(api.ListObjectsArgs)
+		args.Prefix = prefix
+		args.Delimiter = delimiter
+		for {
+			listObjectResult, err := b.client.ListObjects(b.cfg.BucketName, args)
+			if err != nil {
+				return err
+			}
+			for _, content := range listObjectResult.Contents {
+				// LastModified format 2021-10-28T06:55:01Z
+				LastModifiedTime, err := time.Parse(time.RFC3339, content.LastModified)
```

Comment:
convention is `lowerCamelCase` for the local variable

```suggestion
				lastModifiedTime, err := time.Parse(time.RFC3339, content.LastModified)
```